crates/sui-types/src/temporary_store.rs

```diff
@@ -834,10 +912,10 @@ impl<S> TemporaryStore<S> {
         self.deleted.insert(*id, (ctx.clone(), version, kind));
     }
 
-    /// Resets any mutations and deletions recorded in the store.
-    pub fn reset(&mut self) {
-        self.written.clear();
-        self.deleted.clear();
+    /// Resets any mutations and deletions recorded in the store. Ignore object IDs in `retain_ids`
+    pub fn reset(&mut self, retain_ids: BTreeSet<ObjectID>) {
+        self.written.retain(|q, _| retain_ids.contains(q));
+        self.deleted.retain(|q, _| retain_ids.contains(q));
```

Comment:
I'm confused, why do you need to retain any writes?

Reply:
See earlier comment.